lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -84,11 +84,9 @@ MACHINE. For each set of dupes, only the latest job found is included in
 the return array.
 
 =cut
-sub latest_jobs ($self, $until = undef, $limit = undef, $offset = undef) {
-    my %attrs = (order_by => ['me.id DESC']);
-    $attrs{rows} = $limit if defined $limit;
-    $attrs{offset} = $offset if defined $offset;
-    my @jobs = $self->search($until ? {t_created => {'<=' => $until}} : undef, \%attrs);
+
+sub latest_jobs ($self, $until = undef) {
+    my @jobs = $self->search($until ? {t_created => {'<=' => $until}} : undef, {order_by => ['me.id DESC']});
```

Comment:
I know that the code was unused, but just mentioning that when using `latest_jobs($until)` we don't ensure an upper limit. It's not mentioned as an AC but I think we wanted to ensure an upper limit for all queries.

Reply:
That's correct, the `limit` case is explicitly excluded. That was already part of the previous implementation.

Reply:
We could make a followup ticket for figuring out pagination for the `latest` case as well, but it could be pretty hard.

Reply:
Also added a comment to the test related to this.